product_listings_manager/models.py

```diff
@@ -0,0 +1,131 @@
+"""Models definition relecting tables in composedb.
+
+These models are not fully relecting the composedb schema:
+
+- Tables and columns which not needed by PLM are not included.
+
+- Length of db.String() column is not required by postgresql(composedb uses it), in case of
+testing with other db backend(e.g. sqlite) a value is given in following definition and it
+has no side effect to postgresql(composedb).
+"""
+from flask_sqlalchemy import SQLAlchemy
+
+db = SQLAlchemy()
+
+
+class Packages(db.Model):
+    """packages table in composedb.
+
+    Only needed columns in packages table are defined here.
+    """
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(255), nullable=False)
+    arch = db.Column(db.String(32), nullable=False)
+    version = db.Column(db.String(255), nullable=False)
+
+    def __repr__(self):
+        return '<Package %d %s %s %s>' % (self.id, self.name, self.version, self.arch)
+
+
+trees = db.Table(
+    'tree_product_map',
+    db.Column('tree_id', db.Integer, db.ForeignKey('trees.id'), primary_key=True),
+    db.Column('product_id', db.Integer, db.ForeignKey('products.id'), primary_key=True)
+)
+
+
+class Products(db.Model):
+    """products table in composedb."""
+    id = db.Column(db.Integer, primary_key=True)
+    label = db.Column(db.String(100), nullable=False)
+    version = db.Column(db.String(100), nullable=False)
+    variant = db.Column(db.String(200), nullable=False)
```

Comment:
Strictly speaking this column is not nullable (though it probably does not matter in read-only context).

It's bit too late but you can use following command to generate models from existing DB.

```bash
pip install sqlacodegen
sqlacodegen --outfile models.py postgresql://user:passwd@compose-server/compose
```

Reply:
> Strictly speaking this column is not nullable

Sorry, a typo, I meant to say: variant column **is nullable**.

Reply:
Cool tool.